python/spacewalk/satellite_tools/repo_plugins/yum_dnf_src.py

```diff
@@ -199,9 +201,14 @@ def setup_repo(self, repo, no_mirrors, ca_cert_file, client_cert_file, client_ke
 
         if no_mirrors:
             repo.mirrorlist = ""
-        self.digest=hashlib.sha256(self.url.encode('utf8')).hexdigest()[:16]
+        self.digest=hashlib.sha256(dnf_repo_url.encode('utf8')).hexdigest()[:16]
         self.dnfbase.repos.add(repo)
         self.repoid = repo.id
+        headers = tuple()
+        for key in self.http_headers:
+            headers = headers + (key + ": " + self.http_headers[key],)
```

Comment:
[nitpick] If I parse this correctly, another way to write this would be
```python
headers = tuple(f"{name}: {value}" for name, value in self.http_headers.items())
```

I find this a bit more "to the point", but your version is also good.

Reply:
Sorry, I will keep my version in this case. I personally find it easier to read (as a non-developer :) ) and have tested it already.